include/modem/modem_info.h

```diff
@@ -242,6 +242,48 @@ int modem_info_json_object_encode(struct modem_param_info *modem,
  */
 int modem_info_params_get(struct modem_param_info *modem_param);
 
+/** @brief Obtain the UUID of the modem firmware build.
+ *
+ * The UUID is represented as a string, for example:
+ * 25c95751-efa4-40d4-8b4a-1dcaab81fac9
+ *
+ * See the documentation for the AT command %XMODEMUUID.
+ *
+ * @param buf Pointer to the target buffer.
+ *
+ * @retval 0 if the operation was successful.
+ *         Otherwise, a (negative) error code is returned.
+ */
+int modem_info_get_fw_uuid(char *buf);
```

Comment:
I think to input the buffer size too, so that we can have some logic in the functions to prevent buffer overflows